node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,329 @@
+        }
+
+        if (cmeCluster) {
+            await cmeCluster.close();
+        }
+
+        if (managementClient) {
+            managementClient.close();
+        }
+    });
+
+    const runTest = async (
+        test: (client: BaseClient) => Promise<void>,
+        protocol: ProtocolVersion,
+        configOverrides?: Partial<BaseClientConfiguration>,
+    ) => {
+        const isStandaloneMode = configOverrides?.addresses?.length === 1;
+        const activeCluster = isStandaloneMode ? cmdCluster : cmeCluster;
+
+        if (!activeCluster) {
+            throw new Error(
+                `${isStandaloneMode ? "Standalone" : "Cluster"} mode not configured`,
+            );
+        }
+
+        const ClientClass = isStandaloneMode ? GlideClient : GlideClusterClient;
+        const addresses = formatAddresses(activeCluster.getAddresses());
+
+        client = await ClientClass.createClient({
+            addresses,
+            protocol,
+            ...configOverrides,
+        });
+
+        try {
+            await test(client);
+        } finally {
+            client.close();
+        }
+    };
+
+    describe.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "update_connection_password_%p",
+        (protocol) => {
+            const NEW_PASSWORD = "new_password";
+            const WRONG_PASSWORD = "wrong_password";
+
+            /**
+             * Test replacing connection password with immediate re-authentication using a non-valid password.
+             * Verifies that immediate re-authentication fails when the password is not valid.
+             */
+            it("test_update_connection_password_auth_non_valid_pass", async () => {
+                await runTest(async (client: BaseClient) => {
+                    await expect(
+                        client.updateConnectionPassword(null, true),
+                    ).rejects.toThrow(RequestError);
+                    await expect(
+                        client.updateConnectionPassword("", true),
+                    ).rejects.toThrow(RequestError);
+                }, protocol);
+            });
+
+            /**
+             * Test replacing the connection password without immediate re-authentication.
+             * Verifies that:
+             * 1. The client can update its internal password
+             * 2. The client remains connected with current auth
+             * 3. The client can reconnect using the new password after server password change
+             * Currently, this test is only supported for cluster mode,
+             * since standalone mode dont have multiple connections to manage,
+             * and the client will try to reconnect and will not listen to new tasks.
+             */
+            it(
+                "test_update_connection_password",
+                async () => {
+                    await runTest(async (client: BaseClient) => {
+                        if (client instanceof GlideClient) {
```

Comment:
Why skipping stadalone?

Reply:
```
             * Currently, this test is only supported for cluster mode,
             * since standalone mode dont have multiple connections to manage,
             * and the client will try to reconnect and will not listen to new tasks.
    ```

Reply:
Oh shame on me